demo/full/scripts/components/BufferContentGraph.jsx

```diff
@@ -0,0 +1,251 @@
+ * @param {number} height
+ */
+function clearCanvas(canvasContext, width, height) {
+  canvasContext.clearRect(0, 0, width, height);
+}
+
+/**
+ * Scale given bufferedData in terms of percentage between the minimum and
+ * maximum position. Filter out segment which are not part of it.
+ * @param {Array.<Object>} bufferedData
+ * @param {number} minimumPosition
+ * @param {number} maximumPosition
+ * @returns {Array.<Object>}
+ */
+function scaleSegments(bufferedData, minimumPosition, maximumPosition) {
+  const scaledSegments = [];
+  const wholeDuration = maximumPosition - minimumPosition;
+  for (let i = 0; i < bufferedData.length; i++) {
+    const bufferedInfos = bufferedData[i];
+    const start = bufferedInfos.bufferedStart === undefined ?
+      bufferedInfos.start :
+      bufferedInfos.bufferedStart;
+    const end = bufferedInfos.bufferedEnd === undefined ?
+      bufferedInfos.end :
+      bufferedInfos.bufferedEnd;
+    if (end > minimumPosition && start < maximumPosition) {
+      const startPoint = Math.max(start - minimumPosition, 0);
+      const endPoint = Math.min(end - minimumPosition, maximumPosition);
+      const scaledStart = startPoint / wholeDuration;
+      const scaledEnd = endPoint / wholeDuration;
+      scaledSegments.push({ scaledStart,
+                            scaledEnd,
+                            bufferedInfos });
+    }
+  }
+  return scaledSegments;
+}
+
+/**
+ * Display a graph representing what has been buffered according to the data
+ * given.
+ * Allow to seek on click, display the current time, and display a tooltip
+ * describing the buffered data when hovering represented data.
+ * @param {Object}
+ */
+export default function BufferContentGraph({
+  currentTime, // The time currently played
+  data, // The buffered data for a single type of buffer (e.g. "audio")
+  maximumPosition, // The maximum seekable position
+  minimumPosition, // The minimum seekable position
+  seek, // function allowing to seek in the content
+  type, // The type of buffer (e.g. "audio", "video" or "text")
+}) {
+  const [ randomColors ] = useState(shuffleArray(COLORS));
+  const [ tipVisible, setTipVisible ] = useState(false);
+  const [ tipPosition, setTipPosition ] = useState(0);
+  const [ tipText, setTipText ] = useState("");
+  const containerEl = useRef(null);
+  const canvasEl = useRef(null);
+  const representationsEncountered = useRef([]);
+  const duration = Math.max(maximumPosition - minimumPosition, 0);
+
+  const currentSegmentsScaled = useMemo(() => {
+    return scaleSegments(data, minimumPosition, maximumPosition);
+  }, [data, minimumPosition, maximumPosition]);
+
+  useEffect(() => {
+    if (canvasEl === null || canvasEl === undefined) {
+      return ;
+    }
+    const ctx = canvasEl.current.getContext("2d");
+    if (ctx === null) {
+      return;
+    }
+    canvasEl.current.width = CANVAS_WIDTH;
+    canvasEl.current.height = CANVAS_HEIGHT;
+    clearCanvas(ctx, CANVAS_WIDTH, CANVAS_HEIGHT);
```

Comment:
Do you want to clear when no max or min position ?

Reply:
If `minimumPosition` or `maximumPosition` are not defined, it means that we will not be able to represent the x axis (a.k.a. the only axis here).

In that case, I find it logic to clear the canvas from what it could have potentially have kept from a previous draw.

It goes like this:
When one of the variable which possibly impact that graph changed (either `minimumPosition`, `maximumPosition` or `currentSegmentsScaled`):
  1. clear the canvas completely
  2. if `minimumPosition` or `maximumPosition` are not defined or are incoherent (`maximumPosition <= minimumPosition`), we will not be able to represent the buffer's content. In that case, quit - with the canvas empty
  3. Iterate on `currentSegmentsScaled` and represent each one of them in the canvas.

Just to clarify a little thing: `currentSegmentsScaled` is defined from `data` in a `useMemo` as it is the result of a computation used both in a `useEffect` - triggered after the component is rendered on the screen - and in a `mousemove` - triggered when the mouse hovers the canvas.
Defining it in a `useMemo` here allow us to only compute it when one of its dependencies changed (`data`, `minimumPosition` or `maximumPosition`) and to share it between both callbacks in an idiomatic way.